_search-plugins/search-pipelines/index.md

```diff
@@ -0,0 +1,387 @@
+
+## Search response processors
+
+OpenSearch supports the following search response processors:
+
+- [`rename_field`]({{site.url}}{{site.baseurl}}/search-plugins/search-pipelines/rename-field-processor/): Renames an existing field.
+
+## Viewing available processor types
+
+You can use the Nodes Search Pipelines API to view the available processor types:
+
+```json
+GET /_nodes/search_pipelines
+```
+{% include copy-curl.html %}
+
+The response contains the `search_pipelines` object that lists the possible request and response processors:
+
+<details open markdown="block">
+  <summary>
+    Response
+  </summary>
+  {: .text-delta}
+
+```json
+{
+  "_nodes" : {
+    "total" : 1,
+    "successful" : 1,
+    "failed" : 0
+  },
+  "cluster_name" : "runTask",
+  "nodes" : {
+    "36FHvCwHT6Srbm2ZniEPhA" : {
+      "name" : "runTask-0",
+      "transport_address" : "127.0.0.1:9300",
+      "host" : "127.0.0.1",
+      "ip" : "127.0.0.1",
+      "version" : "3.0.0",
+      "build_type" : "tar",
+      "build_hash" : "unknown",
+      "roles" : [
+        "cluster_manager",
+        "data",
+        "ingest",
+        "remote_cluster_client"
+      ],
+      "attributes" : {
+        "testattr" : "test",
+        "shard_indexing_pressure_enabled" : "true"
+      },
+      "search_pipelines" : {
+        "request_processors" : [
+          {
+            "type" : "filter_query"
+          },
+          {
+            "type" : "script"
+          }
+        ],
+        "response_processors" : [
+          {
+            "type" : "rename_field"
+          }
+        ]
+      }
+    }
+  }
+}
+```
+</details>
+
+In addition to the processors provided by OpenSearch, additional processors may be provided by plugins.
+{: .note}
+
+## Creating a search pipeline
+
+Search pipelines are stored in the cluster state. To create a search pipeline, you must configure an ordered list of processors in your OpenSearch cluster. You can have more than one processor of the same type in the pipeline. Each processor has a `tag` identifier that distinguishes it from the others. Tagging a specific processor can be helpful for error messages, especially if you add multiple processors of the same type.
+
+#### Example request
+
+The following request creates a search pipeline with a `filter_query` request processor that uses a term query to return only public messages:
+
+```json
+PUT /_search/pipeline/my_pipeline 
+{
+  "request_processors": [
+    {
+      "filter_query" : {
+        "tag" : "tag1",
+        "description" : "This processor is going to restrict to publicly visible documents",
+        "query" : {
+          "term": {
+            "visibility": "public"
+          }
+        }
+      }
+    }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+## Retrieving search pipelines
+
+To retrieve existing search pipeline details, use the Search Pipeline API. 
```

Comment:
Do we mean "To retrieve the details of an existing search pipeline"? As written, the details are doing the existing.